packages/workspaces-extension/src/index.ts

```diff
@@ -46,13 +50,76 @@ const workspacesMenu: JupyterFrontEndPlugin<void> = {
   }
 };
 
+//Plugin id for workspaces indicator plugin
+const WORKSPACE_INDICATOR_PLUGIN_ID =
+  '@jupyterlab/workspaces-extension:indicator';
+//Command id for toggling workspace indicator
+const WORKSPACE_INDICATOR_COMMAND_ID = 'workspace-indicator:toggle';
+
+/**
+ * The extension providing workspace indicator at topbar
+ */
+const workspacesIndicator: JupyterFrontEndPlugin<void> = {
+  id: WORKSPACE_INDICATOR_PLUGIN_ID,
+  description: 'Adds a workspace indicator element at topbar',
+  requires: [
+    IWorkspacesModel,
+    IWorkspaceCommands,
+    IWindowResolver,
+    ITranslator,
+    ISettingRegistry
+  ],
+  autoStart: true,
+  activate: async (
+    app: JupyterFrontEnd,
+    model: IWorkspacesModel,
+    commands: IWorkspaceCommands,
+    resolver: IWindowResolver,
+    translator: ITranslator,
+    registry: ISettingRegistry
+  ) => {
+    const trans = translator.load('jupyterlab');
+    const openWorkspace = async (workspace: string) => {
+      await app.commands.execute(commands.open, { workspace: workspace });
+    };
+    const workspaceSelector = new WorkspaceSelectorWidget({
+      currentWorkspace: resolver.name,
+      identifiers: model.identifiers,
+      openWorkspace: openWorkspace,
+      model: model,
+      translator: translator
+    });
+    const isToggled = await registry.get(
+      WORKSPACE_INDICATOR_PLUGIN_ID,
+      'toggled'
+    );
+    workspaceSelector.setHidden(!isToggled.composite as boolean);
```

Comment:
This will not update the state if user changes the setting, right?

It should be something as in:


https://github.com/jupyterlab/jupyterlab/blob/5138e10ce54f136b92fb36b9f863653b37ebdb83/packages/shortcuts-extension/src/index.ts#L319-L325

or 

https://github.com/jupyterlab/jupyterlab/blob/5138e10ce54f136b92fb36b9f863653b37ebdb83/packages/apputils-extension/src/palette.ts#L104-L131